esp-wifi/src/wifi/mod.rs

```diff
@@ -609,7 +651,11 @@ unsafe extern "C" fn recv_cb(
     eb: *mut crate::binary::c_types::c_void,
 ) -> esp_err_t {
     let src = core::slice::from_raw_parts_mut(buffer as *mut u8, len as usize);
-    let packet = DataFrame::from_bytes(src);
+    let packet = if let Some(packet) = DataFrame::from_bytes(src) {
+        packet
+    } else {
+        return esp_wifi_sys::include::ESP_ERR_NO_MEM as esp_err_t;
```

Comment:
I think this return may leak memory because `esp_wifi_internal_free_rx_buffer(eb);` is not called.

Reply:
true - we should call `esp_wifi_internal_free_rx_buffer(eb);` here 👍

Reply:
before you do anything, I made a PR this morning that includes this :)